### DISCUSSION TITLE
Get count of directories in sub album

### DISCUSSION THREAD

**Discussion by wizgod** [bpatrik/pigallery2#376] [Q&A] [answered]
Greetings Programs!

In directory.gallery.component.html, I am adding the number of files in the directory to the title:
`<div class="directory-name">{{toTitle(directory.name)}}<span *ngIf='directory.mediaCount > 1'> ({{directory.mediaCount}} files)</span></div>`

I would like to show the number of directories at the same time: "(n albums/n files)"

I was thinking of adding another property to the DirectoryDTO object (directoryCount) but I'm not sure where to set it and how.

Thanks,

Wg

**Comment by wizgod** [comment]
HI @bpatrik ,

Any thoughts on this?

Thanks,

Wg

**Comment by bpatrik** [comment]
Sorry I missed this.
 I added something like this in 231ed38a2aaa311bfde5066af45d9623a6efe199.
 
 Adding album count is more tricky. You will need to create a new filed in the Directory entry and keep up-to-date recursively at every DB update. I'm doing something similar to Album covers.

Note: I personally would prefer not print the number of directories inside a directory. (i.e.: i cannot accept a PR if you planned to so, sorry :/ )

**Reply by bpatrik** [reply]
having a second though, I need to revert 231ed38a2aaa311bfde5066af45d9623a6efe199. Just like with the directories the number of media is also more complex as needed to be counted on the server side.  I'm focusing on more important features now to implement recursive calculation. But showing the number of items is something that I would like to see, as Albums and faces also show it.

**Reply by wizgod** [reply]
Hi Patrik,

I'm just looking to get a count of the directories in the immediate sub folder.

I thought that I could just loop through the directories list and grab the count there. However, I've been going through the backend code and can't seem to find the place to implement this.

Thanks,

Wg

**Comment by wizgod** [comment, marked as answer]
Hi @bpatrik ,

Got it sorted. I've forked it to https://github.com/wizgod/pigallery2 for anyone who might want to do the same.

Comparing to master: https://github.com/bpatrik/pigallery2/compare/master...wizgod:pigallery2:master

Thanks,

Wg
